src/twisted/application/app.py

```diff
@@ -705,4 +706,5 @@ def _exitWithSignal(sig):
     @type sig:  C{int}
     """
     signal.signal(sig, signal.SIG_DFL)
+    atexit._run_exitfuncs()
```

Comment:
This function is internal and undocumented at least as far as the official reference goes.  It seems like the better strategy might be to make your application register a shutdown function with the reactor separately from exitfuncs?

One thing I would be looking for from the documentation here: what are the exception-handling characteristics of this function? Anything we need to catch to make sure `_exitWithSignal` still equally reliably exits with a signal?

Reply:
If an exit handler is registered that raises a new exception then python is expected to return a different exit code

Reply:
Only in Python 2: https://bugs.python.org/issue27035